src/test/java/seedu/address/logic/parser/AddressBookParserTest.java

```diff
@@ -119,4 +120,27 @@ public void parseCommand_owe() throws Exception {
         assertTrue(parser.parseCommand("owe 1 2.50") instanceof OweCommand);
     }
 
+    @Test
+    public void parseCommand_nonAsciiInput_throwsParseException() {
+        String nonAsciiInput = "add José";
+        assertThrows(ParseException.class,
+            AddressBookParser.MESSAGE_NON_ASCII, () -> parser.parseCommand(nonAsciiInput));
+    }
+
+    @Test
+    public void parseCommand_allAsciiInputAllowed() {
+        AddressBookParser parser = new AddressBookParser();
+        for (char c = '!'; c <= '~'; c++) {
```

Comment:
Shouldn't this start with space?

```suggestion
        for (char c = ' '; c <= '~'; c++) {
```

Reply:
Space gets trimmed and gets a different error. Starting with '!' is correct.

Reply:
Changed test case in #f7e1736